rest/views.py

```diff
@@ -212,6 +212,30 @@ def get_queryset(self):
             slug=self.kwargs['vocab_slug']
         )
 
+    def update(self, request, *args, **kwargs):
+        """
+        Override to remove resource term links if resource type is
+        removed. Note that partial_update will call this function too.
+        """
+        vocab = self.get_object()
+        if 'learning_resource_types' in self.request.data:
+            # Since LearningResourceTypes indicate which relationships
+            # are valid between Terms and LearningResources, we need to
+            # remove the newly invalid relationships caused by this update.
+            new_types = set(self.request.data['learning_resource_types'])
+            old_types = set(
+                t.name for t in vocab.learning_resource_types.all()
+            )
+            removed_types = old_types - new_types
+
+            for term in vocab.term_set.all():
+                for resource in term.learning_resources.all():
+                    if resource.learning_resource_type.name in removed_types:
+                        term.learning_resources.remove(resource)
```

Comment:
We do need the reindexing to happen so we can't do this efficiently until the elasticsearch DSL work is in place. But I should add a transaction just in case it fails halfway through